flow360/component/simulation/models/volume_models.py

```diff
@@ -242,6 +249,31 @@ class BETDisk(Flow360BaseModel):
     sectional_polars: List[BETDiskSectionalPolar] = pd.Field()
     sectional_radiuses: List[float] = pd.Field()
 
+    @pd.model_validator(mode="after")
```

Comment:
Why is this an after model validator? Why cannot these be field validators?

If a validation does not require data beyond what is contained in a field/model then do not use after model validator which contains more data than needed.

for example check_bet_disk_alphas_in_order can be:
```python
@pd.field_validator(field="alpha",mode="after")
@classmethod
def check_bet_disk_alphas_in_order(cls,values):
...
```
Having everything in model validator makes it hard to arrange them since pydantic execute them in sequence they were declared.

Reply:
if I use field_validator, I can not access the bet disk's name, which should be included in the error message.

Reply:
https://docs.pydantic.dev/latest/concepts/validators/#field-validators:~:text=If%20you%20want%20to%20access,discussed%20in%20the%20section%20below.